## Ejercicio 1. Estimación de Parámetros Poblacionales: Método de Momentos

El **método de momentos** es una técnica común para estimar parámetros poblacionales a partir de datos muestrales. Se basa en igualar los momentos muestrales (medias, varianzas, etc.) con sus contrapartes poblacionales conocidas.

**Formalización:**

1. **Momentos muestrales:** Se calculan los momentos muestrales de primer, segundo y orden superior a partir de los datos observados.

2. **Ecuaciones de los momentos:** Se establecen ecuaciones que relacionan los momentos muestrales con los parámetros poblacionales desconocidos.

3. **Solución del sistema:** Se resuelve el sistema de ecuaciones para obtener las estimaciones de los parámetros poblacionales.

**Ventajas:**

- Es un método simple y directo que no requiere información adicional sobre la distribución.
- Es útil para estimar parámetros de diversas distribuciones.

**Desventajas:**

- Puede ser menos preciso que otros métodos, especialmente cuando la muestra es pequeña.
- Las ecuaciones de los momentos no siempre son fáciles de resolver.

**Ejemplo 1: Sea una población con distribución normal, encontrar estimadores para los parámetros $\mu$ y
$\sigma^2$, por el método de los momentos, basados en una muestra aleatoria simple de tamaño $n$:**

Si X es una variable aleatoria normal, su función de densidad de probabilidad es:

$f(x;\mu,\sigma^2) = \frac{1}{\sqrt{2\pi}\sigma}e^{-\frac{1}{2}(\frac{x-\mu}{\sigma})^2}$

Al tener dos parámetros ($\mu$ y $\sigma$) se plantearán dos ecuaciones:

**Primer momento poblacional:**

$\mu'_1 = \int_{-\infty}^{\infty}{x\frac{1}{\sqrt{2\pi}\sigma}e^{-\frac{1}{2}(\frac{x-\mu}{\sigma})^2}}dx = \mu$

**Segundo momento poblacional**

$\mu'_2 = \int_{-\infty}^{\infty}{x^2(\frac{1}{\sqrt{2\pi}\sigma}e^{-\frac{1}{2}(\frac{x-\mu}{\sigma})^2})}dx = \sigma^2 + \mu^2$

Por otro lado, los primeros dos momentos muestrales respecto al origen son:

$\mu'_1 = \frac{1}{n}\sum_{i=1}^{n}x_i = \overline{x}$

y

$\mu'_2 = \frac{1}{n}\sum_{i=1}^{n}x_i^2$

Igualando los momentos poblacionales con sus correspondientes momentos muestrales se obtiene un sistema de ecuaciones con dos incógnitas:

$\mu = \overline{x}$

$\sigma^2 + \mu^2 = \frac{1}{n}\sum_{i=1}^{n}x_i^2$

El resultado de la primera ecuación es claro para un estimador de $\mu$. La segunda ecuación se resuelve para la segunda incógnita sustituyendo este valor de la siguiente forma:

$\mu = \overline{x}$

$\sigma^2 + \overline{x}^2 = \frac{1}{n}\sum_{i=1}^{n}x_i^2$

Por lo tanto, los estimadores de los parámetros de una distribución $N(\mu,\sigma^2)$ están dados por la solución a dicho sistema:

$\mu = \overline{x}$

$\sigma^2 = \frac{1}{n}\sum_{i=1}^{n}x_i^2 - \overline{x}^2$

El último resultado se puede reescribir como:

$\sigma^2 = \frac{1}{n}\sum_{i=1}^{n}(x_i - \overline{x})^2$


**Ejemplo 2: Sea una población con distribución Gamma con parámetros $\alpha$ y $\theta$, calcular sus estimadores por el método de momentos.**

Bajo la parametrización $\alpha$ y $\theta$ se tiene la siguiente función de densidad:

$$\displaystyle f_{X}(x)=\begin{cases}
	\displaystyle \frac{x^{\alpha-1}e^{-x/\theta} }{\Gamma(\alpha)\theta^{\alpha}} & x>0\\
	0 & \text{en otro caso}
	\end{cases}$$.

Para el caso de la distribución $Gamma$ es difícil derivar su función de verosimilitud, por lo que es recomendable utilizar el método de momentos para calcular sus estimadores.

**Primer momento poblacional:**

$\mu'_1 = \alpha\theta$

Por otro lado, es más simple utilizar el segundo momento central (respecto la media) para hacer un sistema de ecuaciones simple.

**Segundo momento central:**

$\text{Var}(X) = \alpha\theta^2$

De manera que obtenemos el siguiente sistema de ecuaciones con dos incógnitas:

$E(X) = \alpha\theta = \frac{1}{n} \sum_{i = 1}^{n}x_i = \overline{x}$

$\text{Var}(X) = \alpha\theta^2 = \frac{1}{n} \sum_{i = 1}^{n}(x_i - \overline{x})^2$

Primer resolvemos la primera ecucación para $\alpha$:

$\alpha = \overline{x} / \theta$

Ahora sustituyendo en la segunda ecuación, obtenemos:

$\alpha\theta^2 = \frac{\overline{x}}{\theta}\theta^2 = \overline{x}\theta = \frac{1}{n} \sum_{i = 1}^{n}(x_i - \overline{x})^2$

Finalmente resolviendo para $\theta$:

$\hat{\theta} = \frac{1}{n \overline{x}}\sum_{i = 1}^{n}(x_i - \overline{x})^2$

Sustituímos de nuevo en $\alpha$:

$\hat{\alpha} = \frac{\overline{x}}{\hat{\theta}} = \frac{\overline{x}}{\frac{1}{n \overline{x}}\sum_{i = 1}^{n}(x_i - \overline{x})^2} = \frac{n \overline{x}^2}{\sum_{i = 1}^{n}(x_i - \overline{x})^2}$


**Consideraciones:**

- El método de momentos es más efectivo cuando la muestra es grande y la distribución es conocida.

- Para distribuciones con múltiples parámetros, el sistema de ecuaciones de los momentos puede ser complejo de resolver.

- En algunos casos, se pueden utilizar métodos numéricos para encontrar las soluciones del sistema.


# Ejercicio 2. Calcula el MLE para la media $\mu$ y varianza $\sigma^2$ de una muestra normal.

Sean $x_1,x_2,...,x_n$ una muestra de datos que se asume sigue una distribución normal con media $μ$ y varianza $σ^2$. La distribución normal tiene la siguiente función de densidad de probabilidad:

$$f(x|μ,σ^2)=\frac{1}{\sqrt{2π σ^2}}exp(-\frac{{(x-μ)}^2}{2σ^2})$$

La función de verosimilitud es el producto de las densidades individuales para cada observación:

$$L(\mu, \sigma^2) = \prod_{i=1}^{n} f(x_i|\mu, \sigma^2)$$

Sustituyendo la función de densidad normal, obtenemos:

$$L(\mu, \sigma^2) = \prod_{i=1}^{n} \frac{1}{\sqrt{2\pi\sigma^2}} \exp\left (-\frac{(x_i - \mu)^2}{2\sigma^2}\right)$$

## Simplificación de la Función de Verosimilitud

Para simplificar esta expresión, separamos los términos:

$$L(\mu, \sigma^2) = \left(\frac{1}{\sqrt{2\pi\sigma^2}}\right)^n \exp\left(-\sum_{i=1}^{n} \frac{(x_i - \mu)^2}{2\sigma^2}\right)$$

Se aplica logaritmo natural:

$$\log L(\mu, \sigma^2) = \log \left[ \left(\frac{1}{\sqrt{2\pi\sigma^2}}\right)^n \exp\left(-\sum_{i=1}^{n} \frac{(x_i - \mu)^2}{2\sigma^2}\right) \right]$$

$$=\log \left(\frac{1}{\sqrt{2\pi\sigma^2}}\right)^n + \log \left( \exp\left(-\sum_{i=1}^{n} \frac{(x_i - \mu)^2}{2\sigma^2}\right) \right)$$

$$=n \log \left(\frac{1}{\sqrt{2\pi\sigma^2}}\right) + \left(-\sum_{i=1}^{n} \frac{(x_i - \mu)^2}{2\sigma^2}\right)$$

$$= n \left(-\frac{1}{2} \log (2\pi\sigma^2)\right) - \frac{1}{2\sigma^2} \sum_{i=1}^{n} (x_i - \mu)^2$$

$$= -\frac{n}{2} \log(2\pi\sigma^2) - \frac{1}{2\sigma^2} \sum_{i=1}^{n} (x_i - \mu)^2$$

## Estimación de $μ$

Para encontrar el estimador de $\mu$, derivamos la verosimilitud con respecto a $\mu$ y la igualamos a cero:

La derivada con respecto a $\mu$ es:

$$\frac{\partial \log L(\mu, \sigma^2)}{\partial \mu} = \frac{\partial}{\partial \mu} \left( -\frac{n}{2} \log(2\pi\sigma^2) - \frac{1}{2\sigma^2} \sum_{i=1}^{n} (x_i - \mu)^2 \right)$$

$$= - \frac{1}{2\sigma^2} \frac{\partial}{\partial \mu} \left( \sum_{i=1}^{n} (x_i - \mu)^2 \right)$$

$$= - \frac{1}{2\sigma^2} (-2) \sum_{i=1}^{n} (x_i - \mu) = \frac{1}{\sigma^2} \sum_{i=1}^{n} (x_i - \mu)$$

Igualamos a cero para encontrar el estimador $\hat{\mu}$:

$$\frac{1}{\sigma^2} \sum_{i=1}^{n} (x_i - \mu) = 0$$

$$\sum_{i=1}^{n} x_i - n\mu = 0$$

$$\hat{\mu} = \frac{1}{n} \sum_{i=1}^{n} x_i$$

## Estimación de $σ^2$

Para encontrar el estimador de $\sigma^2$, derivamos la verosimilitud con respecto a $\sigma^2$ y la igualamos a cero:

La derivada con respecto a $\sigma^2$ es:

$$\frac{\partial \log L(\mu, \sigma^2)}{\partial \sigma^2} = \frac{\partial}{\partial \sigma^2} \left( -\frac{n}{2} \log(2\pi\sigma^2) - \frac{1}{2\sigma^2} \sum_{i=1}^{n} (x_i - \mu)^2 \right)$$

Derivando cada término por separado:

$$\frac{\partial}{\partial \sigma^2} \left( -\frac{n}{2} \log(2\pi\sigma^2) \right) = -\frac{n}{2} \cdot \frac{1}{2\sigma^2} \cdot 2\pi = -\frac{n}{2\sigma^2}$$

$$\frac{\partial}{\partial \sigma^2} \left( - \frac{1}{2\sigma^2} \sum_{i=1}^{n} (x_i - \mu)^2 \right) = \sum_{i=1}^{n} (x_i - \mu)^2 \cdot \frac{1}{2\sigma^4}$$

Entonces:

$$\frac{\partial \log L(\mu, \sigma^2)}{\partial \sigma^2} = -\frac{n}{2\sigma^2} + \frac{1}{2\sigma^4} \sum_{i=1}^{n} (x_i - \mu)^2$$

Igualamos a cero para encontrar el estimador $\hat{\sigma}^2$:

$$-\frac{n}{2\sigma^2} + \frac{1}{2\sigma^4} \sum_{i=1}^{n} (x_i - \mu)^2 = 0$$

Multiplicamos ambos lados por $2\sigma^4$:

$$-n\sigma^2 + \sum_{i=1}^{n} (x_i - \mu)^2 = 0$$

$$\sum_{i=1}^{n} (x_i - \mu)^2 = n\sigma^2$$

$$\hat{\sigma}^2 = \frac{1}{n} \sum_{i=1}^{n} (x_i - \hat{\mu})^2$$

#Ejercicio 3
Nos gustaría estimar la proporción de personas que planean votar por el Candidato A en una próxima elección. Se supone que el número de votantes es grande, y $θ$
 es la porción de votantes que planea votar por el Candidato A. Definimos la variable aleatoria $X$ como seguidor. Un votante es elegido uniformemente al azar entre todos los votantes y le preguntamos: "¿Planea votar por el candidato?
 Si dice que "si", entonces $X=1$, en otro caso $X=0$
 Entonces
 $$X\sim Bernoulli(θ)$$
 Sea $X_{1}, X_{2}, X_{3}, ... X_{n}$ una muestra aleatoria de esta distribucion. Es decir que $X_{i}^{´s}$ son i.i.d y $X_{i} \sim Bernoulli(θ)$
 En otras palabras, seleccionamos aleatoriamente n
 votantes (con reemplazo) y le preguntamos a cada uno de ellos si planean votar por el Candidato A. Encuentra un intervalo de confianza $(1- α)100% $ para $θ$ para  $X_{1}, X_{2}, X_{3}, ... X_{n}$

 Dem:
 Sabemos que $Var[X_{i}]=σ^{2}$ y ademas $ θ=E[X_{i}] $
 Entonces $$IC_{α}=[\bar{X} _{n}- Ƶ_{\frac{α}{2}} \frac{σ}{\sqrt{n}},\bar{X} _{n}+ Ƶ_{\frac{α}{2}} \frac{σ}{\sqrt{n}}]$$

 Ahora como es una $Bernoulli(θ)$, su $Var[X_{i}]=θ(1-θ)$ y tambien $Var[X_{i}]=θ(1-θ)=σ^{2} $

Tenemos que buscar la $\sigma$. Pero tenemos que estimar primero $θ$. Entonces si podemos encontrar un limite superior para $\sigma$, podemos usarlo para construir un intervalo de confianza para $θ$.
Para encontrar este limite superior definamos
$$f(θ)=θ(1-θ)$$ para $$ θ\in [0,1]$$
Y ahora usando la derivada e igualando a cero
$f^{'}(\theta)= 1-2\theta$

$0=1-2\theta$

$\theta=\frac{1}{2}$

Entonces el máximo valor $\theta=\frac{1}{2}$

Con lo que $f(\theta)\leq f(\frac{1}{2})= \frac{1}{4}$ para $\theta \in [0,1]$
Así $\sigma_{Max} = \frac{1}{2}$
Entonces concluímos que el intervalo de confianza  $(1- α)100% $ es: $$IC_{α}=[\bar{X} _{n}- Ƶ_{\frac{α}{2}} \frac{ \sigma_{Max}}{\sqrt{n}},\bar{X} _{n}+ Ƶ_{\frac{α}{2}} \frac{\sigma_{Max}}{\sqrt{n}}]$$
$$IC_{α}=[\bar{X} _{n}- Ƶ_{\frac{α}{2}} \frac{1/2}{\sqrt{n}},\bar{X} _{n}+ Ƶ_{\frac{α}{2}} \frac{1/2}{\sqrt{n}}]$$

$$IC_{α}=[\bar{X} _{n}- \frac{Ƶ_{\frac{α}{2}}}{2\sqrt{n}},\bar{X} _{n}+ \frac{Ƶ_{\frac{α}{2}} }{2\sqrt{n}}]$$
Es un intervalo aproximado, sin embargo para números grandes n, es una buena aproximación.

#Ejercicios de la sección 8.3.4
###Problema 2

Una muestra aleatoria $X_{1}$, $X_{2}$, $X_{3}$, ..., $X_{100}$ es dada por una distribución con varianza conocida $Var(X_{i}) = 16$. Para la muestra observada, la media muestral es $\overline{X} = 23.5$. Encuentra una aproximación para un intervalo de confianza del $95\%$ para ${θ} = EX_{i}$.



$Solución$

Tenemos que el siguiente intervalo

$\left[
\overline{X} - z_{\frac{\alpha}{2}} \frac{\sigma}{\sqrt{n}}, \,
\overline{X} + z_{\frac{\alpha}{2}} \frac{\sigma}{\sqrt{n}}
\right]$

es una aproximación para un intervalo de confianza del $(1- {α})100\%$.

Despejando ${α}$ para un intervalo de confianza del $95\%$ tenemos
$95\% = (1-{α})100\%$ entonces, ${α} = 1 - 95\% = 0.05$

Por tanto, tenemos que $z_{\frac{α}{2}} = z_{0.025} = {𝛟}^{-1}(1-0-025)=1.96$

Tambien tenemos que ${\sigma} = 4$. Por lo tanto, la aproximación para el intervalo de confianza es

$\left[
  23.5 - 1.96\frac{4}{\sqrt{100}}, \,
23.5 + 1.96\frac{4}{\sqrt{100}}
\right] \thickapprox [22.7, 24.3].$

Comprobaremos esto usando python:


In [ ]:
import numpy as np
import numpy.random as npr
import scipy.stats as sps
import matplotlib.pyplot as plt
import random
import scipy.stats as st
from scipy.stats import norm

import warnings
warnings.filterwarnings("ignore")

In [ ]:
IC = st.norm.interval(0.95, loc=23.5, scale=.4)
print(IC)

(22.71601440618398, 24.28398559381602)


### Problema 3

Para estimar la proporción de votantes que planean votar por el candidato A en una elección, se utiliza una muestra aleatoria de tamaño $n$ que se elige entre los votantes. El muestreo se realiza con reposición. Sea θ ser la porción de votantes que planean votar por el candidato A entre todos los votantes. ¿Qué tamaño tiene $n$ debe ser para que podamos obtener un 90% intervalo de confianza con 3% ¿margen de error? Es decir, ¿cuán grande es $n$?
tiene que ser tal que

$$
P\left(\overline{X}-0.03 \leq \theta \leq \overline{X}+ 0.03 \right) \geq 0.90,
$$

donde $\overline{X}$ es la porción de personas en nuestra muestra aleatoria que dicen que planean votar por el candidato A.

Aquí:

$$
\left[\overline{X}- \frac{z_{\frac{\alpha}{2}}}{2\sqrt{n}} , \overline{X}+ \frac{z_{\frac{\alpha}{2}}}{2\sqrt{n}}\right]
$$
<br>
es un intervalo de confianza aproximado $(1-\alpha)100\%$.
Como $\alpha = (1-0.90) = 0.1$ tenemos:

<br>

$$
z_{\frac{\alpha}{2}}=z_{\frac{0.1}{2}}=z_{0.05}=\Phi^{-1}(1-0.05)=\Phi^{-1}(0.95)=1.645
$$

necesitamos tener un intervalo de confianza de $3\%$, tenemos que

$$
\frac{1.645}{2\sqrt{n}}=0.03
$$

Por lo tanto, despejamos a $n$ y tenemos

$$
n=\left(\frac{1.645}{2 \times 0.03}\right)^2 = \left(751.673\right) \sim 752
$$

Por lo tanto, se necesita un tamaño de muestra de aproximadamente 752 votantes para lograr un intervalo de confianza del 90% con un intervalo de confianza de 3%.

In [6]:
# calculate z score for 90 percent
import scipy.stats as st
from math import ceil
z = st.norm.ppf(1 - (1 - 0.9) / 2)
print(z)

ME = 0.03 # margen de error

# 0.5 para maximo tamaño de error
p = 0.5

# calculamos el tamaño de la muestra
n = ceil((z/(2*ME))**2)

print("n >=" + str(n))

1.6448536269514722
n >=752


###Problema 4

a. Sea $ X $ una variable aleatoria tal que $\mathbb{R}_X \subseteq [a,bc$, es decir, siempre tenemos $a \leq X \leq b $. Demuestra que

$\text{Var}(X) \leq \frac{(b-a)^2}{4}. \$

b. Sean $ X_1, X_2, X_3, \ldots, X_n$ una muestra aleatoria de una distribución desconocida con función de distribución acumulativa $F_X(x) $ tal que ${R}_X \subseteq [a,b]$. Específicamente, $E(X)$ y $ \text{Var}(X)$ son desconocidos. Encuentra un intervalo de confianza del $(1-\alpha) $100\%  para  $\theta = E(X)$. Supón que \( n \) es grande.

a. Define $Y = \frac{X - a + b}{2}$.Así, ${R}_Y \subseteq \left[ -\frac{b-a}{2}, \frac{b-a}{2} \right]$ \. Entonces,

$\text{Var}(X) = \text{Var}(Y) = E[Y^2] - \mu_Y^2 \leq E[Y^2] \leq \left( \frac{b-a}{2} \right)^2 \quad \text{(ya que } Y^2 \leq \left( \frac{b-a}{2} \right)^2 \text{)} = \frac{(b-a)^2}{4}.$

b. Aquí tenemos un límite superior en $\sigma$, que es $\sigma_{\text{max}} = \left( \frac{b-a}{2} \right)^2$. Por lo tanto, el intervalo
$\left[ \bar{X} - z_{\alpha/2} \frac{\sigma_{\text{max}}}{\sqrt{n}}, \bar{X} + z_{\alpha/2} \frac{\sigma_{\text{max}}}{\sqrt{n}} \right]$
es un intervalo de confianza del $(1-\alpha)\ 100\% \  para \ \theta$. Más específicamente,
$[ \left[ \bar{X} - z_{\alpha/2} \frac{\sqrt{b-a}}{2\sqrt{n}}, \bar{X} + z_{\alpha/2} \frac{\sqrt{b-a}}{2\sqrt{n}} \right]$
es un intervalo de confianza del $(1-\alpha) \ 100\% \ para \ \theta$.

###Problema 5

Se tiene una muestra aleatoria $\quad X_{1}, X_2, X_3, \ldots, X_{144} $ de una distribución con varianza desconocida $\text{Var}(X_i) = \sigma^2$. Para la muestra observada, la media muestral es $\bar{X} = 55.2$ y la varianza muestral es $ S^2 = 34.5 $. Encuentre un intervalo de confianza del 99\% para $\theta = E(X_i)$.

In [ ]:
from scipy.stats import t
alpha = 0.01
df = 144 - 1  # Grados de libertad

# Calcular el cuantil t para alpha/2 = 0.005 y df grados de libertad
t_quantile = t.ppf(1 - alpha/2, df)

print(f"Cuantil t para alpha/2 = {alpha/2} y {df} grados de libertad: {t_quantile:.2f}")

Cuantil t para alpha/2 = 0.005 y 143 grados de libertad: 2.61


El intervalo de confianza del 99\% para la media poblacional $\theta$, basado en la muestra, se calcula como:

$\left[ \bar{X} - Z{\frac{\alpha}{2},} \cdot \frac{S}{\sqrt{n}}, \bar{X} + Z{\frac{\alpha}{2}} \cdot \frac{S}{\sqrt{n}} \right]$

donde:

  * $ \bar{X} = 55.2 $ es la media muestral,

  * $ S = \sqrt{S^2} = \sqrt{34.5} \approx 5.87$ es la desviación estándar muestral,

  * n = 144 es el tamaño de la muestra,

  * $df = n - 1 = 144 - 1 = 143$ son los grados de libertad.

Calculando el cuantil de la distribución t para $ \alpha/2 = 0.005 $ y $df = 143 $:

$ t_{0.005, \, 143} \approx 2.6171$
Sustituyendo los valores:

$ \left[ 55.2 - 2.6171 \cdot \frac{5.87}{\sqrt{144}} \, 55.2 + 2.6171 \cdot \frac{5.87}{\sqrt{144}} \right] \$

Calculamos $( \frac{5.87}{\sqrt{144}} \approx 0.4872 )$:

$\left[ 55.2 - 1.2751 \, 55.2 + 1.2751 \right] \$

$\left[ 53.9249, \, 56.4751 \right]$

Por lo tanto, el intervalo de confianza del 99\% para $\theta$ es aproximadamente $ [53.92, \, 56.48] $.



###Problema 6

A random sample $\quad X_{1}, X_{2}, X_{3}, \ldots, X_{16} $
 is given from a normal distribution with unknown mean $\mu=E[Xi]$
 and unknown variance $Var(Xi)=\sigma^2$
. For the observed sample, the sample mean is $\overline{X}=16.7$
, and the sample variance is $S^2=7.5$
.

Find a 95%
 confidence interval for $\mu$

Find a 95%
 confidence interval for $\sigma^2$

##a. Intervalo de Confianza de Mu
 Aquí, el intervalo


$\left[
\overline{X} - t_{\frac{\alpha}{2}, n-1} \frac{S}{\sqrt{n}}, \,
\overline{X} + t_{\frac{\alpha}{2}, n-1} \frac{S}{\sqrt{n}}
\right]$

es un intervalo de confianza del $(1 - \alpha)100\%$ para $\mu$. Sea $n = 16, \alpha = 0.05$, entonces

$t_{0.025, 15} \approx 2.13$

In [ ]:
from scipy.stats import t


#Calculo de la t0.025,15

# Grados de libertad
df = 15

# Nivel de confianza (95%)
confidence_level = 0.95

# Valor crítico de t
t_critical = t.ppf((1 + confidence_level) / 2, df)

print("Valor crítico de t para un 95% de confianza y 15 grados de libertad:", t_critical)



Valor crítico de t para un 95% de confianza y 15 grados de libertad: 2.131449545559323


In [ ]:
import math

#Calculo del intervalo de confianza para Mu

intervalo_mu = (16.7 - t_critical * (math.sqrt(7.5) / 4)) , (16.7 + t_critical * (math.sqrt(7.5) / 4))
print("Intervalo de confianza para Mu:", intervalo_mu)

Intervalo de confianza para Mu: (15.240696254641279, 18.15930374535872)


Entonces, podemos obtener un intervalo de confianza del $95\%$ para $\mu$ como

$\left[
16.7 - 2.13 \frac{\sqrt{7.5}}{4}, \,
16.7 + 2.13 \frac{\sqrt{7.5}}{4}
\right]
\approx [15.24, 18.16]$

###Por lo tanto, $[15.24, 18.16]$ es un intervalo de confianza del $95\%$ para $\mu$

##b. Intervalo de confianza para sigma cuadrada

Sabemos que, $\left[
\frac{(n-1)S^2}{\chi^2_{\frac{\alpha}{2}, n-1}}, \,
\frac{(n-1)S^2}{\chi^2_{1-\frac{\alpha}{2}, n-1}}
\right]$ es un intervalo de confianza del $(1 - \alpha)100\%$ para $\sigma^2$. En este problema, $n = 16$, $\alpha = 0.05$, así que necesitamos calcular

$\chi^2_{0.025, 15} \approx 27.49, \, \chi^2_{0.975, 15} \approx 6.26$

In [ ]:
from scipy.stats import chi2

#Calculo de las Chi Cuadradas

# Grados de libertad
df = 15

# Nivel de confianza (95%)
confidence_level = 0.95

# Valor crítico de chi-cuadrada en el límite inferior
chi2_lower = chi2.ppf((1 - confidence_level) / 2, df)

# Valor crítico de chi-cuadrada en el límite superior
chi2_upper = chi2.ppf(1 - (1 - confidence_level) / 2, df)

print("Valor crítico de chi-cuadrada para un 95% de confianza y 15 grados de libertad:")
print("Límite inferior:", chi2_lower)
print("Límite superior:", chi2_upper)

Valor crítico de chi-cuadrada para un 95% de confianza y 15 grados de libertad:
Límite inferior: 6.262137795043253
Límite superior: 27.488392863442975


In [ ]:
#Calculo del intervalo de confianza para sigma cuadrada

intervalo_sigma_2 = ((15*7.5)/chi2_upper) , ((15*7.5)/chi2_lower)
print("Intervalo de confianza para Sigma cuadrada es:", intervalo_sigma_2)

Intervalo de confianza para Sigma cuadrada es: (4.092636501481853, 17.965110906541934)


Entonces, podemos obtener un intervalo de confianza del $95\%$ para $\sigma^2$ como

$\left[
\frac{(n-1)S^2}{\chi^2_{0.025, 15}}, \,
\frac{(n-1)S^2}{\chi^2_{0.975, 15}}
\right]=
\left[
\frac{15 \times 7.5}{27.49}, \,
\frac{15 \times 7.5}{6.26}
\right]
\approx [4.09, 17.97]$

###Por lo tanto, $[4.09, 17.97]$ es un intervalo de confianza del $95\%$ para $\sigma^2$